src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/IFhirDataStoreContext.cs

```diff
@@ -0,0 +1,20 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Storage
+{
+    public interface IFhirDataStoreContext
```

Comment:
What is the purpose if this vs CosmosDataStoreConfiguration?

Reply:
When we split the CosmosDB out for Fhir and CP, we end up making the class needing two pieces of information, `CosmosDataStoreConfiguration` and `IOptionsMonitor<CosmosCollectionConfiguration>`, to be able to use `DocumentClient`. Since this information is needed for both `IFhirDataStore` and `IFhirOperationsDataStore`, I just encapsulated that logic into this `IFhirDataStoreContext` class.

If we merge the code back and remove the `IOptionsMonitor<CosmosCollectionConfiguration>`, the we won't need this class anymore.

Reply:
I think we can consolidate.

Reply:
Removed and reverted the code back. There is another work item for tracking merging the Cosmos DB code back.